pallets/identity/src/lib.rs

```diff
@@ -513,6 +513,24 @@ decl_module! {
             Self::join_identity(Signatory::from(sender_did), auth_id)
         }
 
+        /// Leave the signing key's identity.
+        #[weight = SimpleDispatchInfo::FixedNormal(300_000)]
+        pub fn leave_identity_as_key(origin) -> DispatchResult {
+            let sender_key = AccountKey::try_from(ensure_signed(origin)?.encode())?;
+            if let Some(did) = Self::get_identity(&sender_key) {
+                return Self::leave_identity(Signatory::from(sender_key), did);
+            }
```

Comment:
I think we should return an error if we are trying to leave an identity as key, and we are **not** part of this identity or we are part of it _"as identity"_ instead of _"as key"_. (I would use two different error for those cases).

Reply:
`leave_identity` function does throw an error if the signer is not part of the identity. Due to the signed extension requiring every key to have a valid did, the if statement will always be triggered.

When the user calls `leave_identity_as_key` then they are leaving their own identity. "as identity" does not make sense in this case. Let's say alice_key is part of acme_did and calls this function then she is trying to leave acme_did. I do not see the point in checking if acme_did is a signer of acme_did.